### PR TITLE
Add registered stake pool owners to database schema

### PR DESCRIPTION
**Pull request by rvl**
# Issue Number

Relates to #1091.

# Overview

- Adds database table for stake pool owners
- Adds sqlite queries for adding owners and querying by pool id.
- Adds model/mvar implementation (fwiw)
- Property test combinines the put and read functions.

# Comments

This will cause a database migration, which is adding a new table.

### CODE REVIEW

File: lib/core/src/Cardano/Pool/DB/Model.hs [hunk 1]
```diff
@@ -146,6 +153,20 @@ mReadStakeDistribution epoch db@PoolDatabase{distributions} =
     , db
     )
 
+mPutStakePoolOwner :: PoolId -> PoolOwner -> ModelPoolOp ()
+mPutStakePoolOwner poolId poolOwner db@PoolDatabase{owners} =
+    ( Right ()
+    , db { owners = (entry:filter (/= entry) owners) }
```
Comment (KtorZ) [inline lib/core/src/Cardano/Pool/DB/Model.hs]: **[remark]**

Alternatively, could use a `Set` for this.

File: lib/core/src/Cardano/Pool/DB/Sqlite.hs [hunk 2]
```diff
@@ -163,6 +163,15 @@ newDBLayer logConfig trace fp = do
                 [ StakeDistributionEpoch ==. fromIntegral epoch ]
                 []
 
+        , putStakePoolOwner = \poolId poolOwner -> do
+            deleteWhere [PoolOwnerPoolId ==. poolId, PoolOwnerOwner ==. poolOwner]
+            insert_ (PoolOwner poolId poolOwner)
```
Comment (KtorZ) [inline lib/core/src/Cardano/Pool/DB/Sqlite.hs]: **[suggestion]**

Use [`repsert`](https://hackage.haskell.org/package/persistent-2.10.1/docs/Database-Persist-Class.html#v:repsert). 

**Why?**

Better convey the semantic behind these two "statements", and slightly more concise.

File: lib/core/test/unit/Cardano/Pool/DB/Arbitrary.hs [hunk 3]
```diff
@@ -98,6 +107,14 @@ instance Arbitrary PoolId where
         bytes <- vectorOf 32 (elements ['a'..'z'])
         return $ PoolId $ B8.pack bytes
 
+instance Arbitrary StakePoolOwnersFixture where
+    arbitrary = StakePoolOwnersFixture <$> fmap PoolId bs <*> listOf entry
+      where
+        entry = (,) <$> fmap PoolId bs <*> fmap PoolOwner bs
+        bs = B8.replicate 32 <$> elements ['a'..'e']
```
Comment (KtorZ) [inline lib/core/test/unit/Cardano/Pool/DB/Arbitrary.hs]: **[suggestion]**

rename `bs` to `genBs` or `genBytes` 

**Why?**

At first glance, a reader may interpret `bs` as a byte string. Yet, it is indeed a generator here and therefore leads to some confusion. Prefixing with `gen`  better reflect what the let binding captures.

File: lib/core/test/unit/Cardano/Pool/DB/Properties.hs [hunk 4]
```diff
@@ -347,6 +351,32 @@ prop_putStakePutStake DBLayer {..} epoch a b =
...
+prop_stakePoolOwner DBLayer {..} (StakePoolOwnersFixture poolId entries) =
+    monadicIO (setup >> prop)
+  where
+    setup = run $ atomically cleanDB
+    prop = do
+        run . atomically $ mapM_ (uncurry putStakePoolOwner) entries
+        res <- run $ atomically (readStakePoolOwners poolId)
+        monitor $ classify dups "duplicate entries"
+        monitor $ classify (haveMany snd) "multiple owners per pool"
+        monitor $ classify (haveMany fst) "multiple pools per owner"
```
Comment (KtorZ) [inline lib/core/test/unit/Cardano/Pool/DB/Properties.hs:367]: :+1:

**Review by KtorZ — APPROVED** [review]
LGTM. Some minor remarks and tweaks .

File: lib/core/src/Cardano/Pool/DB/Model.hs (hunk 1 above)
Comment (rvl) [reply lib/core/src/Cardano/Pool/DB/Model.hs]: True, fixed.

File: lib/core/src/Cardano/Pool/DB/Sqlite.hs (hunk 2 above)
Comment (rvl) [reply lib/core/src/Cardano/Pool/DB/Sqlite.hs]: I considered repsert, but it also needed a Key argument which was meh.
But I've changed it now.

File: lib/core/test/unit/Cardano/Pool/DB/Arbitrary.hs (hunk 3 above)
Comment (rvl) [reply lib/core/test/unit/Cardano/Pool/DB/Arbitrary.hs]: OK fixed.

**Pull request merged**